app/questionnaire/navigator.py

```diff
@@ -104,54 +19,74 @@ def __init__(self, survey_json, metadata=None, answer_store=None):
         self.first_block_id = SchemaHelper.get_first_block_id(self.survey_json)
```

Comment:
We don't use this first_block_id other than below to create the first_block_location, maybe we should get rid as we have lots of first location related fields